lib/clients/AzureClient.py

```diff
@@ -0,0 +1,453 @@
+                disk_name = 'sf-disk-{}'.format(
+                    time.strftime("%Y%m%d%H%M%S"))
+                disk_creation_operation = self.compute_client.disks.create_or_update(
+                    self.resource_group,
+                    disk_name,
+                    {
+                        'location': snapshot.location,
+                        'disk_size_gb': size,
+                        'creation_data': {
+                            'create_option': DiskCreateOption.empty
+                        },
+                        'account_type': StorageAccountTypes.standard_lrs
+                    }
+                )
+
+            self._wait('Waiting for volume {} to get ready...'.format(disk_name),
+                       lambda operation: operation.done() is True,
+                       None,
+                       disk_creation_operation)
+
+            disk = disk_creation_operation.result()
+            volume = Volume(disk.name, 'none', disk.disk_size_gb)
+            self._add_volume(volume.id)
+            self.logger.info(
+                '{} SUCCESS: volume-id={}'.format(log_prefix, volume.id))
+        except Exception as error:
+            message = '{} ERROR: size={}\n{}'.format(log_prefix, size, error)
+            self.logger.error(message)
+            if volume:
+                self.delete_volume(volume.id)
+                volume = None
+            raise Exception(message)
+
+        return volume
+
+    def _delete_volume(self, volume_id):
+        log_prefix = '[VOLUME] [DELETE]'
+
+        try:
+            disk_deletion_operation = self.compute_client.disks.delete(
+                self.resource_group, volume_id)
+
+            self._wait('Waiting for volume {} to be deleted...'.format(volume_id),
+                       lambda operation: operation.done() is True,
+                       None,
+                       disk_deletion_operation)
+            delete_response = disk_deletion_operation.result()
+            self._remove_volume(volume_id)
+            self.logger.info(
+                '{} SUCCESS: volume-id={}\n{}'.format(
+                    log_prefix, volume_id, delete_response))
+            return True
+        except Exception as error:
+            message = '{} ERROR: volume-id={}\n{}'.format(
+                log_prefix, volume_id, error)
+            self.logger.error(message)
+            raise Exception(message)
+
+    def _create_attachment(self, volume_id, instance_id):
+        log_prefix = '[ATTACHMENT] [CREATE]'
+        attachment = None
+
+        try:
+            virtual_machine = self.compute_client.virtual_machines.get(
+                self.resource_group,
+                instance_id
+            )
+            volume = self.compute_client.disks.get(
+                self.resource_group, volume_id)
+            all_data_disks = virtual_machine.storage_profile.data_disks
+            # ideally should be travers through all disks and find next balnk lun
```

Comment:
comment spelling mistake

Reply:
actually the instruction in comment is implemented .. updated the comment.